### ISSUE TITLE
Cart Bundle Test App Intended Functionality

### ISSUE THREAD

**Issue by devinatbryt** [Shopify/function-examples#223]
###### *Note: this is while using "Shopify's Bundle App", not one I've built.*
I was trying to get the price adjustment working as the cart bundled items don't actually give you access to the bundle variants, you just have reference to them which is the ID. In doing so I noticed that when I add an item that is a part of the bundle, it combines it into the bundle. But if I also add the actual bundle variant, the following happens:
1. ![image](https://user-images.githubusercontent.com/101842591/233744127-c14f351c-9a8d-44a9-a6f2-002bec2b35ee.png)
2. ![image](https://user-images.githubusercontent.com/101842591/233743875-69b6cf7d-6b6e-4cb3-96cf-a733eff59fd9.png)

Furthermore when I go back to the cart, it now appears as if I have two of the same items in cart:
![image](https://user-images.githubusercontent.com/101842591/233744277-d584a9d6-7669-44ca-89ff-5d4d1c5ca1ec.png)

###### *The following is just a brief overview of the above metafields.*
For the "broken fire" product, I've defined the following metafields as such:
![image](https://user-images.githubusercontent.com/101842591/233744856-8a1c14ca-26e5-4aa1-9d99-cb3198f3be27.png)

For the "Bundle Test" variant (which is the parent component), I have the following values/metafields attached to the product:
![image](https://user-images.githubusercontent.com/101842591/233745017-98210f45-d213-4e67-bb73-203dd930e9d7.png)

## Questions
1. **Does the user have to add the bundle parent product, the bundled children or both in order for the bundle to appear?**
2. **What should the user flow look like in order to add the bundled product into the cart?**

#### Final Note:
I personally would like the ability to add just the bundle product to cart and it show the individual line items within the expand operation feature. That way the total is accurate to what the bundle is price

Furthermore, it's weird that going back breaks the cart as it actually mutates the item in the cart to be the bundled items. This causes confusion from the user as they'd have to remove all 4 items that are apart of the bundle instead of just the 1 bundled item. 

###### *Thank you for taking a look into this*

**Comment by tman16** [comment]
Same issue here.

I also wasn't sure if the Cart page is supposed to be showing the Bundle dropdown list with each of the bundled components just like on the Checkout page that would be useful also but hate having to use Rust....need a fully operation js template

**Comment by devinatbryt** [comment]
> 

Hey, I actually ended up converting one of their other js extension examples over to the cart extension and it works perfectly. I'll provide more details about this soon. But essentially Shopify created a tool called javy that allows you to transpile js to web assembly. When I have time later today, I'll post a few examples of it working and how the files and structure look 😎.

**Comment by devinatbryt** [comment]
> @devinatbryt I have this exact same problem; I want the parent product to remain, not the child...but I am getting this same result from the app bundle sample app.

Yeah, It's also important to note that these functions aren't yet available for production sites. So there may be things they change in the future. It's been a while since I've come back to this. Although it'd definitely be nice to get a response from a Shopify rep about this issue 😅.

**Comment by tman16** [comment]
> > Same issue here.
> > I also wasn't sure if the Cart page is supposed to be showing the Bundle dropdown list with each of the bundled components just like on the Checkout page that would be useful also but hate having to use Rust....need a fully operation js template
> 
> Hey, I actually ended up converting one of their other js extension examples over to the cart extension and it works perfectly. I'll provide more details about this soon. But essentially Shopify created a tool called javy that allows you to transpile js to web assembly. When I have time later today, I'll post a few examples of it working and how the files and structure look, although it may not be appropriate to post it here. I'll reference the GitHub comment I made about this exact same question.

With this did you still manage to get the cart transform to work on the Cart page after going back from checkout though?

Also it seems that the extension points for the UI in particular "Admin::Product::Configuration::Render" isn't even working yet so I am not sure how anyone has managed to create a working app for the 30th June. It's already been extended from the end of May so wouldn't be surprised if pushed back another month or more. Looking like Bundles isn't a priority and just going to have to wait...
